fe/fe-core/src/main/java/com/starrocks/qe/scheduler/slot/GlobalSlotProvider.java

```diff
@@ -0,0 +1,184 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.qe.scheduler.slot;
+
+import com.starrocks.common.Config;
+import com.starrocks.common.Status;
+import com.starrocks.common.UserException;
+import com.starrocks.common.util.DebugUtil;
+import com.starrocks.ha.LeaderInfo;
+import com.starrocks.qe.scheduler.RecoverableException;
+import com.starrocks.rpc.FrontendServiceProxy;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.thrift.TNetworkAddress;
+import com.starrocks.thrift.TReleaseSlotRequest;
+import com.starrocks.thrift.TReleaseSlotResponse;
+import com.starrocks.thrift.TRequireSlotRequest;
+import com.starrocks.thrift.TStatusCode;
+import com.starrocks.thrift.TUniqueId;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.thrift.TApplicationException;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+/**
+ * Manage slot globally in the whole cluster when enabling query queue.
+ * <p> It is the slot manager view in the follower FEs. It receives the slot operations from
+ * {@link com.starrocks.qe.scheduler.Coordinator} and sends it to {@link SlotManager} via RPC.
+ *
+ * @see SlotManager
+ */
+public class GlobalSlotProvider implements SlotProvider {
```

Comment:
This file is changed nothing but renamed from `SlotProvider` to `GlobalSlotProvider`.